File: Language AI Azure/Azure_CS-App/lib/models/message.dart
class Message {
  final String id;
  final String text;
  final String? sentiment;
  final double? positive;
  final double? neutral;
  final double? negative;
  final int? idDataset;

  Message({
    required this.id,
    required this.text,
    this.sentiment,
    this.positive,
    this.neutral,
    this.negative,
    this.idDataset,
  });

  Message.empty({
    this.id = '',
    this.text = '',
    this.sentiment,
    this.positive,
    this.neutral,
    this.negative,
    this.idDataset,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    if (json['positivePercentage'].runtimeType == int) {
      json['positivePercentage'] = json['positivePercentage'].toDouble();
    }

    if (json['neutralPercentage'].runtimeType == int) {
      json['neutralPercentage'] = json['neutralPercentage'].toDouble();
    }

    if (json['negativePercentage'].runtimeType == int) {
      json['negativePercentage'] = json['negativePercentage'].toDouble();
    }

    return Message(
      id: json['id'],
      text: json['messageContent'],
      sentiment: json['messageSentiment'],
      positive: json['positivePercentage'],
      neutral: json['neutralPercentage'],
      negative: json['negativePercentage'],
      idDataset: json['idDataset'],
    );
  }
}

File: Language AI Azure/AzureApi (Example)/models/Message.cs
namespace AzureApi.models;

public class Message
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string MessageContent { get; set; } = string.Empty;
    public string? MessageSentiment { get; set; }
    public double? PositivePercentage { get; set; }
    public double? NeutralPercentage { get; set; } 
    public double? NegativePercentage { get; set; }
    public int? IdDataset { get; set; }
}